app/controllers/users_controller.rb

```diff
@@ -93,9 +94,15 @@ def request_destroy
 
   def confirm_destroy
     destroy_token = Rails.cache.read("user-destroy-token-#{@user.id}")
-    raise ActionController::RoutingError, "Not Found" unless destroy_token.present? && destroy_token == params[:token]
 
-    set_current_tab("account")
+    # rubocop:disable Layout/LineLength
+    if destroy_token.blank?
+      flash[:settings_notice] = "Your token has expired, please request a new one. Tokens only last for 12 hours after account deletion is initiated."
+      redirect_to user_settings_path("account") and return
+    end
+    # rubocop:enable Layout/LineLength
+
+    raise ActionController::RoutingError, "Not Found" unless destroy_token == params[:token]
```

Comment:
Makes sense, to avoid using the `and return` we might want to restructure the if (haven'te tested) into:

```ruby
if destroy_token.blank?
  redirect
elsif destroy_token != params[:token]
  raise
end

Reply:
When I had tried this previously, Rubocop did _not_ like the conditional written that way, but I just tried this again and it worked, so I've made this change! 😊